### PR TITLE
WIP  Add http authentication handling

### PR DESCRIPTION
**Pull request by copejon**

### CODE REVIEW

File: cmd/importer/importer.go [hunk 1]
```diff
@@ -27,21 +26,13 @@ func main() {
...
-		glog.Fatalf("main(): \n")
-	}
-	dataReader, err := NewDataReader(importInfo)
+	dataStream, err := NewDataStreamFactory(ep, acc, sec).NewDataStream()
 	if err != nil {
-		glog.Fatalf("main: unable to create data reader: %v\n", err)
+		glog.Fatalf("main: error getting data stream: %v", err)
 	}
-	defer dataReader.Close()
-	glog.Infof("Beginning import from %s\n", importInfo.Url.RawPath)
```
Comment (jeffvance) [inline cmd/importer/importer.go]: Do we still want the func name (e.g. _main:_) in err / log msgs? Most of the msgs here include the func name but maybe that's temp for initial debugging?

File: pkg/importer/dataStreamFactory.go [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+	}
+	if d.url.Scheme == "s3" { // TODO what about non-aws s3 interfaces? (minio just performs http GET on them anyway)
+		return d.s3()
+	}
+	return d.http()
+}
+
+func (d *dataStreamFactory) s3() (io.ReadCloser, error) {
+	glog.Infoln("Using S3 client to get data")
+	bucket := d.url.Host
```
Comment (jeffvance) [inline pkg/importer/dataStreamFactory.go]: ditto: func name not in msg

File: pkg/importer/dataStreamFactory.go [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+		return nil, d.err
+	}
+	if d.url.Scheme == "s3" { // TODO what about non-aws s3 interfaces? (minio just performs http GET on them anyway)
+		return d.s3()
+	}
+	return d.http()
+}
+
+func (d *dataStreamFactory) s3() (io.ReadCloser, error) {
+	glog.Infoln("Using S3 client to get data")
```
Comment (jeffvance) [inline pkg/importer/dataStreamFactory.go]: ditto: func name not in msg

**Comment by jeffvance** [comment]
+1 on restructuring

File: cmd/importer/importer.go (hunk 1 above)
Comment (copejon) [reply cmd/importer/importer.go]: We probably should.  My method up to now was to only do it in errors, but it's wise to mark all log output that way.  Too bad glog doesn't do that for us.  There are _other_ logging packages that will... But I don't think we want to vendor in some random guy's 3rd party package.

File: pkg/common/common.go [hunk 4]
```diff
@@ -4,7 +4,7 @@ package common
 
 const (
 	// host file constants:
-	IMPORTER_WRITE_DIR     = "/data"
+	IMPORTER_WRITE_DIR     = "."
```
Comment (copejon) [inline pkg/common/common.go]: oops

**Comment by jeffvance** [comment]
LGTM.
Minor nit: there are still some glog.Info msgs with inconsistent text formatting. Eg some mgs are missing their func name.

**Pull request merged**